syncer/syncer.go

```diff
@@ -1049,6 +1052,17 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		if err != nil {
 			return err
 		}
+
+		// for fresh and all-mode task, flush checkpoint so we could delete metadata file
+		if s.cfg.Mode == config.ModeAll {
+			if err := s.flushCheckPoints(); err != nil {
+				s.tctx.L().Warn("fail to flush checkpoints when starting task", zap.Error(err))
+			} else {
+				s.tctx.L().Info("try to remove loaded files")
+				os.Remove(path.Join(s.cfg.Dir, "metadata"))
```

Comment:
it seems that this line is not necessary because will remove the dir next line

Reply:
`os.Remove` could only remove an empty directory.

Reply:
use `RemoveAll`?

Reply:
afraid that there're other files, so I just remove specified files.

Reply:
👌